pkg/operator/certrotation/target.go

```diff
@@ -121,6 +121,7 @@ func (c RotatedSelfSignedCertKeySecret) EnsureTargetCertKeyPair(ctx context.Cont
 
 	if reason := c.CertCreator.NeedNewTargetCertKeyPair(targetCertKeyPairSecret, signingCertKeyPair, caBundleCerts, c.Refresh, c.RefreshOnlyWhenExpired, creationRequired); len(reason) > 0 {
 		c.EventRecorder.Eventf("TargetUpdateRequired", "%q in %q requires a new target cert/key pair: %v", c.Name, c.Namespace, reason)
+		c.AdditionalAnnotations.RefreshPeriod = c.Refresh.String()
 		if err = setTargetCertKeyPairSecretAndTLSAnnotations(targetCertKeyPairSecret, c.Validity, signingCertKeyPair, c.CertCreator, c.AdditionalAnnotations); err != nil {
```

Comment:
i would also pass `c.Refresh` to `setTargetCertKeyPairSecretAndTLSAnnotations` and then to `setTLSAnnotationsOnTargetCertKeyPairSecret` so that annotation assignments are in one place for the target.